client/packages/invoices/src/Prescriptions/LineEditView/PrescriptionLineEditForm.tsx

```diff
@@ -70,39 +77,24 @@ export const PrescriptionLineEditForm: React.FC<
   isAutoAllocated,
   hasOnHold,
   hasExpired,
+  isLoading,
+  updateQuantity,
 }) => {
   const t = useTranslation();
   const [allocationAlerts, setAllocationAlerts] = useState<StockOutAlert[]>([]);
-  const [issueQuantity, setIssueQuantity] = useState(0);
+  const [issueUnitQuantity, setIssueUnitQuantity] = useState(0);
   const { format } = useFormatNumber();
   const { rows: items } = usePrescription();
 
-  const onChangePackSize = (newPackSize: number) => {
-    const packSize = newPackSize === -1 ? 1 : newPackSize;
-    const newAllocatedQuantity =
-      newPackSize === 0 ? 0 : Math.round(allocatedQuantity / packSize);
-
-    packSizeController.setPackSize(newPackSize);
-    allocate(newAllocatedQuantity, newPackSize);
-  };
-
-  const updateIssueQuantity = (quantity: number) => {
-    setIssueQuantity(
-      Math.round(
-        quantity / Math.abs(Number(packSizeController.selected?.value || 1))
-      )
-    );
-  };
-
   const debouncedSetAllocationAlerts = useDebounceCallback(
     warning => setAllocationAlerts(warning),
     []
   );
 
-  const allocate = (quantity: number, packSize: number) => {
+  const allocate = (numPacks: number, packSize: number) => {
     const newAllocateQuantities = onChangeQuantity(
-      quantity,
-      packSize === -1 ? null : packSize,
+      numPacks,
+      packSize === -1 || packSize === 1 ? null : packSize,
```

Comment:
This is particularly clunky, but seems to be what it needs to work with the current auto-allocate method. I'd like to change that to be more understandable overall, but it effects other stock issuing areas too, so not worth risking breaking them at this point.

Reply:
I had a quick look at this, looks like -1 comes from the `Any` option for allocation in pack size controller? Which shouldn't ever be set to `-1` in our case then??

Probably worth creating a refactor issue to clean this up, but agree should be a new PR & issue. 

However I think if packSize is 1, then it would be ok to pass null? Did you see a problem with that?
I'm fine with how it is just wanted to understand what was going on :)

Reply:
I think it's fine, I just found it really hard to understand why you need to pass `null` when the packSize is 1, but the actual value for the others. It's not very intuitive...